### ISSUE TITLE
Stream Error & Replace By New Connection Handling help

### ISSUE THREAD

**Issue by vickyrappier** [robbiehanson/XMPPFramework#646]
Hi Team /Robbie,

I am usually getting this 

`2015-12-18 18:18:34:047 Closrr_DV4[14510:9327] RECV: <stream:error xmlns:stream="http://etherx.jabber.org/streams"><policy-violation xmlns="urn:ietf:params:xml:ns:xmpp-streams"/><text xmlns="urn:ietf:params:xml:ns:xmpp-streams" lang="">session in resume state cannot accept incoming stanzas</text></stream:error>`

Also how to handle Stream Replaced by New Connection as when this happens then the Previous chats (few) and receipts we sent get missed.

Appreciate if anyone can highlight what we have to do in this cases.

**Comment by vickyrappier** [comment]
Should we also Enable the background mode for the app when using XMPP Framework as most of the cases many presence / other stanza are received in logs even when app in in background (IOS9) and so we unable to detect the stuff we are handling correctly or not.

**Comment by vickyrappier** [comment]
Hello Robbie & Folks, 

Can you please help in this what we need and researched is following.

1) Every few minutes the client sends/receives Stream replaced by new connection In Presence and all areas. But the new Stream is not able to handle messaging and hence the message packets are lost. Which are either  send  or received.

2) We saw as per XEP-0198 there is something to deal with the Resumption of connection.

Example 10 in (http://xmpp.org/extensions/xep-0198.html)
C: <resume xmlns='urn:xmpp:sm:3' 
           h='some-sequence-number'
           previd='some-long-sm-id'/>

S: <resumed xmlns='urn:xmpp:sm:3' 
            h='another-sequence-number'
            previd='some-long-sm-id'/>

I see there is a function in XMPPFramwork as below but this is never called.

/**
 * Internal method that handles sending the <resume/> element, and the corresponding state transition.
**/
- (void)sendResumeRequestWithResumptionId:(NSString *)resumptionId
{
}

Can you quickly advice please on this.

**Comment by bintu1234** [comment]
I'm also facing saming did you find any solution

**Comment by bintu1234** [comment]
@vickyrappier if you find any solution please help me I'm Struck from last two days.

**Comment by Jethvasagar** [comment]
I found some details :
The server either (1) is closing the existing stream for this entity because a new stream has been initiated that conflicts with the existing stream, or (2) is refusing a new stream for this entity because allowing the new stream would conflict with an existing stream (e.g., because the server allows only a certain number of connections from the same IP address or allows only one server-to-server stream for a given domain pair as a way of helping to ensure in-order processing

for more details please follow this link [streams-error-conditions](https://xmpp.org/rfcs/rfc6120.html#streams-error-conditions)

**Comment by FidelisChimombe** [comment]
I agree with Jethvasagar's findings. I experienced the same error, then I realized that I was initiating a new xmpp connection in an infinite loop, as a result a newly established stream was being closed when a subsequent new connection was established, as a result there were conflicts between closing and opening new connection.
